src/lighteval/tasks/registry.py

```diff
@@ -142,22 +124,89 @@ def get_task_dict(
             TASKS_TABLE.extend(module.TASKS_TABLE)
 
         if len(TASKS_TABLE) > 0:
-            custom_tasks_registry = create_config_tasks(meta_table=TASKS_TABLE, cache_dir=self.cache_dir)
+            custom_tasks_registry = create_config_tasks(meta_table=TASKS_TABLE, cache_dir=self._cache_dir)
             hlog(custom_tasks_registry)
 
+        default_tasks_registry = create_config_tasks(cache_dir=self._cache_dir)
+        # Check the overlap between default_tasks_registry and custom_tasks_registry
+        intersection = set(default_tasks_registry.keys()).intersection(set(custom_tasks_registry.keys()))
+        if len(intersection) > 0:
+            hlog_warn(
+                f"Following tasks ({intersection}) exists both in the default and custom tasks. Will use the default ones on conflict."
+            )
+
+        # Defaults tasks should overwrite custom tasks
+        return {**default_tasks_registry, **custom_tasks_registry}
+
+    @property
+    @lru_cache
+    def _task_superset_dict(self):
+        """
+        Returns a dict mapping each superset to its tasks:
+        eg: mmlu -> [mmlu:abstract_algebra, mmlu:college_biology, ...]
+        """
+
+        superset_dict = {k: list(v) for k, v in groupby(list(self.task_registry.keys()), lambda x: x.split(":")[0])}
+        # Only consider supersets with more than one task
+        return {k: v for k, v in superset_dict.items() if len(v) > 1}
+
+    @property
+    @lru_cache
+    def task_groups_dict(self) -> dict[str, str]:
+        """
+        Returns a dict mapping each task group to its tasks
+        eg: all_cusotm -> custom|task1,custom|task2,custom|task3
```

Comment:
```suggestion
        eg: all custom tasks -> custom|task1,custom|task2,custom|task3
```
Note: the first param (custom/helm/lighteval/...) is not a "task group" but a suite the task comes from, initially/a way of implementing the task now. I think the docstring should reflect this for consistency.

Reply:
If this actually refers to task groups (`custom|mmlu:subtask1,cutom|mmlu:subtask2`) example must be updated accordingly